### PR TITLE
Add Facebook login

### PR DESCRIPTION
**Pull request by federicoaldunate**
Add Facebook login with 'omniauth-facebook' gem and connect it to devise.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -68,6 +68,7 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 #Authentication
 gem 'devise'
+gem 'omniauth-facebook'
```
Comment (jiv) [inline Gemfile]: Specify the gem version

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,27 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable, :confirmable
+         :recoverable, :rememberable, :trackable, :validatable, :confirmable, :omniauthable,
```
Comment (Alxzu) [inline app/models/user.rb]: I'll suggest encapsulating this logic (also the `self.from_omniauth(auth)`) as a concern, [see this example](https://medium.com/@carlescliment/about-rails-concerns-a6b2f1776d7d).

File: app/models/user.rb [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+      user.name = auth[:name]
+      user.gender = match_gender(auth[:gender])
+      user.skip_confirmation!
+    end
+  end
+
+  def self.match_gender(gender)
+    0 if (gender != 'male') && (gender != 'female')
+    gender
+  end
```
Comment (alealvarez1) [inline app/models/user.rb]: I would change it to: 
```
0 if %w(male female).exclude? gender
gender
```

File: app/controllers/users/omniauth_callbacks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        'uid': auth.uid,
+        'email': data.email,
+        'name': data.name,
+        'gender': match_gender(auth.extra.raw_info.gender),
+        'password': Devise.friendly_token[0, 20]
+      }
+    end
+
+    def match_gender(gender)
+      0 if %w[male female].exclude? gender
```
Comment (GuilleLeopold) [inline app/controllers/users/omniauth_callbacks_controller.rb]: This methods seems to just be returning `gender`, if you want to return gender or 0, you should add a `return` here

**Pull request merged**